### PR TITLE
IGNITE-17588 SQL API for C++ Client

### PR DESCRIPTION
**Pull request by isapego**

### CODE REVIEW

File: modules/platforms/cpp/ignite/protocol/utils.cpp [hunk 1]
```diff
@@ -28,20 +28,73 @@
...
 namespace ignite::protocol {
 
+/**
+ * Check if int value fits in @c T.
+ *
+ * @tparam T Int type to fit value to.
+ * @param value Int value.
+ */
+template<typename T>
+inline void check_int_fits(std::int64_t value) {
```
Comment (ptupitsyn) [inline modules/platforms/cpp/ignite/protocol/utils.cpp:38]: This looks suspicious. I think we can use MsgPack type code prefixes to check if the next value in the stream is `int` or something else.

File: modules/platforms/cpp/tests/client-test/sql_test.cpp [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+#include <gtest/gtest.h>
+
+#include <chrono>
+
+using namespace ignite;
+
+/**
+ * Test suite.
+ */
+class sql_test : public ignite_runner_suite {
```
Comment (ptupitsyn) [inline modules/platforms/cpp/tests/client-test/sql_test.cpp:32]: Let's add more tests for various failure conditions, such as:
* Invalid SQL text - does the user get a sensible error message?
* What if I iterate the result set twice?
* Can I access result set pages after closing the cursor?

**Review by ptupitsyn — CHANGES_REQUESTED** [review]

File: modules/platforms/cpp/tests/client-test/sql_test.cpp (hunk 2 above)
Comment (isapego) [reply modules/platforms/cpp/tests/client-test/sql_test.cpp:32]: That's a good proposal. I'll add more tests.

Comment (isapego) [reply modules/platforms/cpp/tests/client-test/sql_test.cpp:33]: Added tests

**Review by ptupitsyn — APPROVED** [review]

**Pull request merged**